admin/tool/policy/classes/api.php

```diff
@@ -396,4 +402,118 @@ public static function move_up($policyid) {
     public static function move_down($policyid) {
         static::move_policy_document($policyid, 3);
     }
+
+    /**
+     * Returns list of acceptances
+     *
+     * @param int|array $users id of a user or a list of ids
+     * @param int|array $versions list of policy versions
+     * @return array list of acceptances indexed by userid and versionid. Each acceptance has additional field 'policyid'
+     */
+    public static function get_acceptances($versions, $select = '', $params = [], $orderby = 'u.lastname, u.firstname', $limitfrom = 0, $limitto = 0) {
+        global $DB;
+        $versions = is_array($versions) ? array_values($versions) : [$versions];
+        list($vsql, $vparams) = $DB->get_in_or_equal($versions, SQL_PARAMS_NAMED, 'ver');
+        $userfields = get_all_user_name_fields(true, 'u', null, 'user');
+        $userfieldsmod = get_all_user_name_fields(true, 'm', null, 'mod');
+        $sql = "SELECT u.id AS mainuserid, $userfields, $userfieldsmod, a.*, u.policyagreed
+                  FROM {user} u
+                  LEFT OUTER JOIN {tool_policy_acceptances} a ON a.userid = u.id AND a.policyversionid $vsql
+                  LEFT OUTER JOIN {user} m ON m.id = a.usermodified";
```

Comment:
Just a personal preference, I would not include the `OUTER` keyword here. We do not typically do it in Moodle code base and it only gives a false impression of significance where there is none really.

Reply:
I agree, my query did not work (for another reason) and I added "outer" hoping it would fix it. I forgot to remove it :)

Reply:
Thanks :+1: